lib/middleware.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+var middleware = require('express').Router();
+
+module.exports = function generateMiddleware(options) {
+
+  options = options || {};
+
+  options['cookie-name'] = options['cookie-name'] || 'hof-cookie-check';
+
+  middleware.get('/cookies-required', function renderCookiesRequired(req, res) {
+    res.render('cookies-required');
+  });
+
+  middleware.get('/check-cookies', function checkCookie(req, res) {
+    if (Object.keys(req.cookies).length && req.cookies[options['cookie-name']]) {
+      res.redirect(req.cookies[options['cookie-name']]);
```

Comment:
req.originalUrl is not the full url with domain it's as follows:

```js
app.use('/admin', function(req, res, next) {
  // GET 'http://www.example.com/admin/new'
  console.log(req.originalUrl); // '/admin/new'
  console.log(req.baseUrl); // '/admin'
  console.log(req.path); // '/new'
  next();
});
```

I guess we could encrypt the cookie. Do we care? I'm not sure it offers much... @timgent whadda ya think?